lib/rubocop/cop/rspec/describe_class.rb

```diff
@@ -44,12 +44,20 @@ class DescribeClass < Cop
 
         def_node_matcher :shared_group?, SharedGroups::ALL.block_pattern
 
-        def on_top_level_describe(node, args)
+        def on_top_level_describe(node, (described_value, _))
           return if shared_group?(root_node)
           return if valid_describe?(node)
           return if describe_with_rails_metadata?(node)
+          return if string_constant_describe?(described_value)
 
-          add_offense(args.first)
+          add_offense(described_value)
+        end
+
+        private
+
+        def string_constant_describe?(described_value)
+          described_value.str_type? &&
+            described_value.value =~ /^(?:(?:::)?[A-Z]\w*)+$/
```

Comment:
Deep research [here](https://stackoverflow.com/questions/53549477/regex-to-test-criteria-for-class-name), check it out.

This regex matched `'Active_Record::Base'`, I first thought it shouldn't, but it seems to be a valid class name though.

Is `?:` necessary? You can use `.match?` instead of `=~` to get a boolean.

Reply:
> This regex matched `'Active_Record::Base'`, I first thought it shouldn't, but it seems to be a valid class name though.

Yep, that's why I leave this.

---

> Is `?:` necessary?

No. I thought it's faster, but not:

```ruby
# frozen_string_literal: true

require 'bundler/setup'
Bundler.setup

require 'pry-byebug'

require 'benchmark'
require 'benchmark/ips'
require 'benchmark/memory'


puts '```ruby'
puts File.read(__FILE__).gsub("\t", '  ')
puts '```'
puts
puts '### Output'
puts
puts '```'


STRING = 'Active_Record::Base'


def with_non_captured_group
  STRING =~ /^(?:(?:::)?[A-Z]\w*)+$/
end

def with_captured_group
  STRING =~ /^(?:(?:::)?[A-Z]\w*)+$/
end


def test
  results = %i[
    with_non_captured_group
    with_captured_group
  ].each_with_object([]) do |method_name, array|
    array << (result = send method_name)
    puts "Result of #{method_name}:"
    pp result
    puts
  end
  uniq_size = results.uniq.size
  puts "Uniq results: #{uniq_size}"
  puts
  exit 1 if uniq_size > 1
end

test

Benchmark.ips do |x|
  x.report('with_non_captured_group') { with_non_captured_group }
  x.report('with_captured_group') { with_captured_group }

  x.compare!
end

Benchmark.memory do |x|
  x.report('with_non_captured_group') { 100.times { with_non_captured_group } }
  x.report('with_captured_group') { 100.times { with_captured_group } }

  x.compare!
end


puts '```'
```

### Output

```
Result of with_non_captured_group:
0

Result of with_captured_group:
0

Uniq results: 1

Warming up --------------------------------------
with_non_captured_group
                       185.982k i/100ms
 with_captured_group   183.201k i/100ms
Calculating -------------------------------------
with_non_captured_group
                          2.463M (± 3.0%) i/s -     12.461M in   5.064546s
 with_captured_group      2.430M (± 2.7%) i/s -     12.274M in   5.055925s

Comparison:
with_non_captured_group:  2462742.5 i/s
 with_captured_group:  2429607.0 i/s - same-ish: difference falls within error

Calculating -------------------------------------
with_non_captured_group
                        16.800k memsize (     0.000  retained)
                       100.000  objects (     0.000  retained)
                         0.000  strings (     0.000  retained)
 with_captured_group    16.800k memsize (     0.000  retained)
                       100.000  objects (     0.000  retained)
                         0.000  strings (     0.000  retained)

Comparison:
with_non_captured_group:      16800 allocated
 with_captured_group:      16800 allocated - same
```

Removed.

---



> You can use `.match?` instead of `=~` to get a boolean.

I wanted, but I saw `spec.required_ruby_version = '>= 2.3.0'`, and `String#match?` was introduced in Ruby 2.4.

Reply:
Fair enough 👍